docs/troubleshooting-guide/performance.md

```diff
@@ -0,0 +1,110 @@
+
+4. Otherwise, if the question was created using the notebook editor, we translate the MBQL representation of the question into SQL. Storing an abstract representation of the question and translating it on the fly lets us create SQL specific to the database you're using: while SQL is supposed to be a standard, its various implementations differ in many ways.
+
+5. The back end then sends the request to the database. It needs a connection to the database in order to do this; like most applications, Metabase maintains a pool of connections that it re-uses rather than making a fresh connection each time.
+
+6. The database executes the query and returns the result to the back end.
+
+7. The back end forwards those results to the front end.
+
+8. The front end creates the HTML to display them.
+
+## Specific Problems
+
+Each of the steps described is a potential performance bottleneck:
+
+### The Metabase instance is getting so much traffic that loading the HTML page is slow.
+
+This is pretty rare, since our front end is not very large and browsers cache our JavaScript, but it's easy to rule out.
+
+**How to detect this:** Nothing appears on the page (not even the controls), or elements appear one by one.
+
+**How to fix this:** Look at page load times in the browser console [FIXME: link]. If it's taking a long time to load our HTML, CSS, or JavaScript, check to see whether a proxy, firewall, or other network component is slowing things down.
+
+### Caching is disabled.
+
+By default caching is disabled so that we always re-run every question. However, if your data is only being updated every few seconds or minutes, you will improve performance by enabling caching.
+
+**How to detect this:** Open the Admin Panel, go to "Settings", and look in the "Caching" tab to see whether caching is enabled or not.
+
+**How to fix this:** [This guide][admin-caching] explains how to change the minimum query duration (we cache anything that takes longer than that to run) and the maximum cache size (which is the total memory the cache will ever use). You may need to experiment with these values over several days to find the best balance. For example, if you increase the total memory used by the cache, you may also want to increase the memory that the JVM is allowed to use (discussed below).
+
+### The answer you want isn't cached.
+
+If you have just restarted Metabase, it  has to refresh all its caches, so performance on some questions will be slower the first time you view them. This only happens once per question, but if you have thousands of users, each with their own custom questions, you may see a lot of slowdowns occurring all at once.
+
+**How to detect this:** Look at Metabase's logs, or in the server's logs, to see when it was last restarted.
+
+**How to fix this:** Patience is a virtue.
+
+### The database is overloaded by other traffic.
```

Comment:
Could easily be caused by someone opening a dashboard with 20+ long-running queries, which will then mean everyone else trying to query the database will have to wait until there's available connections to the database. This is outlined in the scaling article.

Reply:
incorporated.